src/storage/src/storage/signed_url.rs

```diff
@@ -14,8 +14,30 @@
 
 use crate::{error::SigningError, signed_url::UrlStyle};
 use auth::signer::Signer;
+use percent_encoding::{AsciiSet, utf8_percent_encode};
 use std::collections::BTreeMap;
-use tokio::time::Duration;
+
+/// https://cloud.google.com/storage/docs/request-endpoints#encoding
+/// !, #, $, &, ', (, ), *, +, ,, /, :, ;, =, ?, @, [, ]
+const PATH_ENCODE_SET: AsciiSet = AsciiSet::EMPTY
```

Comment:
This is suspiciously similar to:

https://github.com/googleapis/google-cloud-rust/blob/0b42085164397827ea1a5705c63bb4380960289d/src/storage/src/storage/client.rs#L699-L718

We probably want to share both definitions.  And I am 99.44% sure you want to percent-encode `\n` and `\r` (which are legal characters in an object name) too.